src/scmrepo/git/backend/pygit2.py

```diff
@@ -412,6 +417,53 @@ def push_refspecs(
     ) -> Mapping[str, SyncStatus]:
         raise NotImplementedError
 
+    def _merge_remote_branch(
+        self,
+        rh: str,
+        lh: str,
+        force: bool = False,
+        on_diverged: Optional[Callable[[str, str], bool]] = None,
+    ) -> SyncStatus:
+        import pygit2
+
+        rh_rev = self.resolve_rev(rh)
+
+        if force:
+            self.set_ref(lh, rh_rev)
+            return SyncStatus.SUCCESS
+
+        try:
+            merge_result, _ = self.repo.merge_analysis(rh_rev, lh)
+        except KeyError:
+            self.set_ref(lh, rh_rev)
+            return SyncStatus.SUCCESS
+
+        if merge_result & pygit2.GIT_MERGE_ANALYSIS_UP_TO_DATE:
+            return SyncStatus.UP_TO_DATE
+        if merge_result & pygit2.GIT_MERGE_ANALYSIS_FASTFORWARD:
+            self.set_ref(lh, rh_rev)
+            return SyncStatus.SUCCESS
+        if merge_result & pygit2.GIT_MERGE_ANALYSIS_NORMAL:
+            if on_diverged and on_diverged(lh, rh_rev):
+                return SyncStatus.SUCCESS
+            return SyncStatus.DIVERGED
+        logger.debug("Unexpected merge result: %s", pygit2.GIT_MERGE_ANALYSIS_NORMAL)
+        raise SCMError("Unknown merge analysis result")
+
+    @contextmanager
+    def get_remote(self, url: str) -> Generator["Remote", None, None]:
+        try:
+            yield self.repo.remotes[url]
+        except ValueError:
+            remote_name = uuid()
+            self.repo.remotes.set_url(remote_name, url)
+            try:
+                yield self.repo.remotes[remote_name]
```

Comment:
This still has the same issue where you will call `set_url` on a remote that doesn't actually exist which will likely end up causing errors and/or undefined behavior in the underlying libgit C library.

If you are going to implement it this way and create a full named remote (instead of adding the anonymous remote function call to pygit), you need to be using `self.repo.remotes.create()` here